Sources/HTTP/HTTPServer.swift

```diff
@@ -26,15 +26,33 @@ public protocol HTTPServing : class {
 /// abstraction, but the intention is to remove this dependency and reimplement
 /// the class using transport APIs provided by the Server APIs working group.
 public class HTTPServer: HTTPServing {
+
+    /// A subclass for HTTP Options
+    /// HTTPServer to be created on the given `port`, using `handler` to process incoming requests
+    public class Options {
+        /// Note: if port 0 is requested, a random port will be assigned and HTTPServer.port value will
+        /// diverge from HTTPServer.Options.port
+        public let port: Int
+        public let handler: HTTPRequestHandler
+
+        ///  Create an instance of HTTPServerOptions
+        public init(onPort: Int = 0, withHandler: @escaping HTTPRequestHandler) {
+            port = onPort
+            handler = withHandler
+        }
+    }
+    public let options: Options
+    
     private let server = PoCSocketSimpleServer()
 
     /// Create an instance of the server. This needs to be followed with a call to `start(port:handler:)`
-    public init() {
+    public init(with newOptions: Options) {
```

Comment:
I think this should be `init(handler: .., options:)`. The `with` style is for method names, not for init (I think, not sure).

Reply:
For the names, I followed the patterns of dataTask.
`dataTask(with:)`
https://developer.apple.com/documentation/foundation/urlsession/1411554-datatask
`dataTask(with:completionHandler:)`
https://developer.apple.com/documentation/foundation/urlsession/1410330-datatask